src/pybind/hmm/hmm_topology_pybind.cc

```diff
@@ -0,0 +1,104 @@
+// pybind/hmm/hmm_topology_pybind.cc
+
+// Copyright 2020   Mobvoi AI Lab, Beijing, China
+//                  (author: Fangjun Kuang, Yaguang Hu, Jian Wang)
+
+// See ../../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "hmm/hmm_topology_pybind.h"
+
+#include "hmm/hmm-topology.h"
+
+using namespace kaldi;
+
+void pybind_hmm_topology(py::module& m) {
+  using PyClass = HmmTopology;
+  auto hmm = py::class_<PyClass>(
+      m, "HmmTopology",
+      "A class for storing topology information for phones. See `hmm` for "
+      "context. This object is sometimes accessed in a file by itself, but "
+      "more often as a class member of the Transition class (this is for "
+      "convenience to reduce the number of files programs have to access).");
+
+  using State = HmmTopology::HmmState;
+  py::class_<State>(
+      hmm, "HmmState",
+      "A structure defined inside HmmTopology to represent a HMM state.")
+      .def(py::init<>())
+      .def(py::init<int>(), py::arg("pdf_class"))
+      .def(py::init<int, int>(), py::arg("forward_pdf_class"),
+           py::arg("self_loop_pdf_class"))
+      .def_readwrite("forward_pdf_class", &State::forward_pdf_class)
```

Comment:
Is it really necessary to add bindings down to this detailed level?

Reply:
I intended to expose such APIs to Python so that what we can do in C++, we can also do it in Python. I haven't thought of a situation where such low-level bindings can be useful.

I can remove such bindings now and add them when they are really needed.

Reply:
Its OK, was just checking, we can keep them.